### PR TITLE
Integrates check, checkmate, and stalemate checking into command-line Game loop

### PR DESCRIPTION
**Pull request by alxmjo**

### CODE REVIEW

File: source/Game.cpp [hunk 1]
```diff
@@ -6,35 +6,88 @@
...
-
 		while (ss >> buf)
 		{
 			tokens.push_back(buf);
 		}
-
 		std::pair<int, int> from = board.algebraicToInt(tokens[0]); 	// convert substring to pair
 		std::pair<int, int> to = board.algebraicToInt(tokens[1]);		// convert substring to pair
 
+		// Verify that the correct player is moving
```
Comment (djbauman) [inline source/Game.cpp]:
```suggestion
		// Verify that the correct player is moving
		
		if (board.getPiece(from) != NULL)
		{
			if (board.getPiece(from)->getColor() != toMove)
			{
				std::cout << "Error: It's " << printColor(toMove) << "'s turn." << '\n';
				continue;
			}
		}
		else
		{
			std::cout << "Error: No piece to move." << "\n";
		}
```

File: source/Game.cpp [hunk 2]
```diff
@@ -6,35 +6,88 @@
...
+		if (board.getPiece(from)->getColor() != toMove)
```
Comment (djbauman) [inline source/Game.cpp]:
```suggestion
		if (board.getPiece(from) != NULL)
		{
			if (board.getPiece(from)->getColor() != toMove)
			{
				std::cout << "Error: It's " << printColor(toMove) << "'s turn." << '\n';
				continue;
			}
		}
		else
		{
			std::cout << "Error: No piece to move." << "\n";
		}
```

Comment (djbauman) [reply source/Game.cpp]: Something like this, to allow me to input an invalid move without throwing a nullptr exception.

File: source/Game.cpp [hunk 3]
```diff
@@ -6,35 +6,88 @@
...
+		{
+			std::cout << "Error: It's " << printColor(toMove) << "'s turn." << '\n';
+			continue;
+		}
+
+		// Move piece
 		board.movePiece(from, to);
```
Comment (djbauman) [inline source/Game.cpp]:
```suggestion
		if (board.movePiece(from, to))
		{
			// Verify that move doesn't put player in check
			if (isInCheck(toMove))
			{
				// If move puts player in check, print error, revert move, and let player enter different move
				std::cout << "Error: This puts " << printColor(toMove) << " in check.\n";
				board.revertLastMove();
				continue;
			}

			// Switch players
			currentMove++;
			toMove = getTurn(currentMove);
		}
		else
		{
			std::cout << "Error: invalid move. Still " << printColor(toMove) << "'s turn.\n";
		}
```

Comment (djbauman) [reply source/Game.cpp]: Something like this to prevent the turn from advancing when an invalid move is attempted.

File: source/Game.cpp (hunk 2 above)
Comment (alxmjo) [reply source/Game.cpp]: Let me know if my most recent commit doesn't resolve this for you. I never had any exceptions, but I also never put in malformed commands. Obviously this has no error checking whatsoever, so anything that's outsides bounds of accepted input will crash program.

File: source/Game.cpp (hunk 3 above)
Comment (alxmjo) [reply source/Game.cpp]: Used basically all of this. Thank you!

**Comment by alxmjo** [comment]
Please have a look at my most recent commit (95687f2...).

**Pull request merged**